tools/cli/workflowCommands.go

```diff
@@ -298,74 +297,69 @@ func formatInputsForDisplay(inputs []interface{}) string {
 }
 
 func unmarshalSearchAttrFromCLI(c *cli.Context) map[string]interface{} {
-	sanitize := func(val string) []string {
-		trimmedVal := strings.TrimSpace(val)
-		if len(trimmedVal) == 0 {
-			return nil
-		}
-		splitVal := strings.Split(trimmedVal, searchAttrInputSeparator)
-		result := make([]string, len(splitVal))
-		for i, v := range splitVal {
-			result[i] = strings.TrimSpace(v)
-		}
-		return result
+	if !c.IsSet(FlagSearchAttributesKey) && c.IsSet(FlagSearchAttributesValue) {
+		ErrorAndExit(fmt.Sprintf("Search attribute keys must be provided using %s.", FlagSearchAttributesKey), nil)
 	}
 
-	searchAttrKeys := sanitize(c.String(FlagSearchAttributesKey))
-	if len(searchAttrKeys) == 0 {
-		return nil
-	}
-	rawSearchAttrVals := sanitize(c.String(FlagSearchAttributesVal))
-	if len(rawSearchAttrVals) == 0 {
+	saKeys := c.StringSlice(FlagSearchAttributesKey)
+
+	var saValues []string
+	if c.IsSet(FlagSearchAttributesValue) {
+		saValues = c.StringSlice(FlagSearchAttributesValue)
+	} else if c.IsSet(FlagSearchAttributesKey) {
+		ErrorAndExit(fmt.Sprintf("Search attribute values must be provided using %s.", FlagSearchAttributesValue), nil)
+	} else {
```

Comment:
These 2 cases should be combined with the first if check of this func to make this more readable.

```
isKeySet := c.IsSet(FlagSearchAttributesKey)
isValueSet := c.IsSet(FlagSearchAttributesValue)
if !isKeySet && !isValueSet {
  return nil
} else if isKeySet && !isValueSet {
  ErrorAndExit(...)
} else if !isKeySet && isValueSet {
  ErrorAndExit(...)
}
saKeys := c.StringSlice(FlagSearchAttributesKey)
saValues = c.StringSlice(FlagSearchAttributesValue)
...
```